tests/suites/test_suite_cipher.function

```diff
@@ -120,6 +123,13 @@ void cipher_special_behaviours( )
     TEST_ASSERT( mbedtls_cipher_set_iv( &ctx, iv, 0 )
                  == MBEDTLS_ERR_CIPHER_BAD_INPUT_DATA );
 
+    mbedtls_cipher_free( &ctx );
```

Comment:
I think that when `MBEDTLS_CIPHER_MODE_CBC` is defined, then we will call `mbedtls_cipher_free()` and then `mbedtls_cipher_setup()` without calling `mbedtls_cipher_init()` in between...

Reply:
perhaps, but I think it's more intuitive to have a `free`\\`init` pair.
For example, if we want to add mutex to the context , we would like to initialize the mutex after we free it

Reply:
@RonEld: I think I agree, but I am not entirely sure I understand your comment. I definitely think that free/init should be called in pairs, so the point of my comment is that when `MBEDTLS_CIPHER_MODE_CBC` is defined you have the following sequence which does not respect that idea of pairs:
```
mbedtls_cipher_init()
//// Some work
mbedtls_cipher_free()
//// Some work
mbedtls_cipher_free()
```

Reply:
Right, I missed the `mbedtls_cipher_init()` for some reason. I'll fix that